### ISSUE TITLE
Naming Conflicts in Clojure 1.11

### ISSUE THREAD

**Issue by skylee03** [clojure-finance/datajure#16]
Starting from Clojure 1.11, running the test program will generate the following warnings:

```
WARNING: update-vals already refers to: #'clojure.core/update-vals in namespace: clojure.tools.analyzer.utils, being replaced by: #'clojure.tools.analyzer.utils/update-vals
WARNING: update-keys already refers to: #'clojure.core/update-keys in namespace: clojure.tools.analyzer.utils, being replaced by: #'clojure.tools.analyzer.utils/update-keys
WARNING: update-vals already refers to: #'clojure.core/update-vals in namespace: clojure.tools.analyzer, being replaced by: #'clojure.tools.analyzer.utils/update-vals
WARNING: update-keys already refers to: #'clojure.core/update-keys in namespace: clojure.tools.analyzer, being replaced by: #'clojure.tools.analyzer.utils/update-keys
WARNING: update-vals already refers to: #'clojure.core/update-vals in namespace: clojure.tools.analyzer.passes, being replaced by: #'clojure.tools.analyzer.utils/update-vals
WARNING: update-vals already refers to: #'clojure.core/update-vals in namespace: clojure.tools.analyzer.passes.uniquify, being replaced by: #'clojure.tools.analyzer.utils/update-vals
WARNING: random-uuid already refers to: #'clojure.core/random-uuid in namespace: onyx.static.uuid, being replaced by: #'onyx.static.uuid/random-uuid
WARNING: random-uuid already refers to: #'clojure.core/random-uuid in namespace: onyx.peer.task-compile, being replaced by: #'onyx.static.uuid/random-uuid
WARNING: random-uuid already refers to: #'clojure.core/random-uuid in namespace: onyx.peer.read-batch, being replaced by: #'onyx.static.uuid/random-uuid
WARNING: random-uuid already refers to: #'clojure.core/random-uuid in namespace: onyx.peer.peer-group-manager, being replaced by: #'onyx.static.uuid/random-uuid
WARNING: random-uuid already refers to: #'clojure.core/random-uuid in namespace: onyx.peer.communicator, being replaced by: #'onyx.static.uuid/random-uuid
WARNING: random-uuid already refers to: #'clojure.core/random-uuid in namespace: onyx.api, being replaced by: #'onyx.static.uuid/random-uuid
```

This is because Clojure 1.11 introduced the `update-keys` and `update-vals` functions.

Update: Solved by upgrading Clojask.

**Comment by skylee03** [comment]
Solved by adding the following JVM option in `project.clj`:

```clojure
:jvm-opts ["--add-opens=java.base/sun.nio.ch=ALL-UNNAMED"]
```

**Comment by skylee03** [comment]
Reference: https://stackoverflow.com/questions/73465937/apache-spark-3-3-0-breaks-on-java-17-with-cannot-access-class-sun-nio-ch-direct

**Comment by skylee03** [comment]
```
WARNING: abs already refers to: #'clojure.core/abs in namespace: zero-one.geni.core.functions, being replaced by: #'zero-one.geni.core.functions/abs
```

Fixed in Geni but it is not released.

https://github.com/zero-one-group/geni/pull/342

**Comment by skylee03** [comment]
Now can be fixed by upgrading Geni to v0.0.41 (to be released soon).

**Comment by skylee03** [comment]
The problem is not completely solved in v0.0.41. `src/clojure/zero_one/geni/core/functions.clj` should also be updated.

**Comment by skylee03** [comment]
Current solution:
- Downgrade Clojure to v1.10.3
- Users can still specify newer versions in their projects

**Comment by skylee03** [comment]
> The problem is not completely solved in v0.0.41. `src/clojure/zero_one/geni/core/functions.clj` should also be updated.

It will be solved soon!
- https://github.com/zero-one-group/geni/issues/348
- https://github.com/zero-one-group/geni/pull/352
